File: cpp_pubsub/src/publisher_member_function.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <string>

#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
#include "tutorial_interfaces/msg/num.hpp"

using namespace std::chrono_literals;


class MinimalPublisher : public rclcpp::Node {
    public:
        MinimalPublisher() : Node("minimal_publisher"), count_(0)
        {
            timer_ = this->create_wall_timer(500ms, std::bind(&MinimalPublisher::TimerCallback, this));
            publisher_ = this->create_publisher<tutorial_interfaces::msg::Num>("topic",10);
        }
    private:
        void TimerCallback(){
            auto message = tutorial_interfaces::msg::Num();
            message.num = count_++;
            RCLCPP_INFO(this->get_logger(),"Publishing '%ld'", message.num);
            publisher_->publish(message);
        }
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr publisher_;
    size_t count_;
};

int main(int argc, char *argv[]){
    rclcpp::init(argc,argv);
    rclcpp::spin(std::make_shared<MinimalPublisher>());
    rclcpp::shutdown();
    return 0;
}

File: py_pubsub/py_pubsub/publisher_member_function.py
import rclpy
from rclpy.node import Node

from std_msgs.msg import String

class MinimalPublisher(Node):
    def __init__(self):
        super().__init__("minimal_publisher")
        self.publisher_ =  self.create_publisher(String, "topic", 10)
        time_period = 0.5
        self.timer = self.create_timer(time_period, self.timer_callback)
        self.count = 0

    def timer_callback(self):
        message = String()
        message.data = "Hello World '%d' " % self.count
        self.publisher_.publish(message)
        self.get_logger().info("Published '%s'" % message.data)
        self.count += 1

def main(args=None):
    rclpy.init(args=args)
    minimal_publisher = MinimalPublisher()
    rclpy.spin(minimal_publisher)
    minimal_publisher.destroy_node()
    rclpy.shutdown()

if __name__ == "__main__":
    main()